### ISSUE TITLE
LLDP Malformed Packet

### ISSUE THREAD

**Issue by aperezmollo** [raspi/WinLLDPService#11]
### Submission type

  - [x] Bug report
  - [ ] Request for enhancement

### Used version of operating system
 Windows 7 Professional 64 bit

### In case of bug report: Is machine connected to Active Directory?

 - [x] Yes, Windows 2012

### In case of bug report: Version the issue has been seen with

17.2.9.2046

### In case of bug report: Expected behaviour you didn't see
Attach possible log(s) and screenshot(s).

```
Frame 5965: 424 bytes on wire (3392 bits), 424 bytes captured (3392 bits) on interface 0
    Interface id: 0 (\Device\NPF_{4BF60FC5-443B-4922-94B2-D9593843D225})
    Encapsulation type: Ethernet (1)
    Arrival Time: Feb 14, 2017 14:45:57.096556000 Hora estándar de Argentina
    [Time shift for this packet: 0.000000000 seconds]
    Epoch Time: 1487094357.096556000 seconds
    [Time delta from previous captured frame: 0.004609000 seconds]
    [Time delta from previous displayed frame: 8.220908000 seconds]
    [Time since reference or first frame: 25.589902000 seconds]
    Frame Number: 5965
    Frame Length: 424 bytes (3392 bits)
    Capture Length: 424 bytes (3392 bits)
    [Frame is marked: False]
    [Frame is ignored: False]
    [Protocols in frame: eth:ethertype:lldp]
    [Coloring Rule Name: Broadcast]
    [Coloring Rule String: eth[0] & 1]
Ethernet II, Src: Dell_71:66:db (5c:26:0a:71:66:db), Dst: LLDP_Multicast (01:80:c2:00:00:0e)
    Destination: LLDP_Multicast (01:80:c2:00:00:0e)
    Source: Dell_71:66:db (5c:26:0a:71:66:db)
    Type: 802.1 Link Layer Discovery Protocol (LLDP) (0x88cc)
Link Layer Discovery Protocol
    Chassis Subtype = MAC address, Id: 5c:26:0a:71:66:db
    Port Subtype = Locally assigned, Id: Conexi\303\263n de \303\241rea local
        0000 010. .... .... = TLV Type: Port Id (2)
        .... ...0 0001 1001 = TLV Length: 25
        Port Id Subtype: Locally assigned (7)
        Port Id: Conexi\357\277\275\357\277\275n de \357\277\275\357\277\275rea local
    Time To Live = 120 sec
    Port Description = Name:Conexi?n de ?rea local | GW:10.212.12.1 | NM:0, 22 | DNS:10.105.50.21,  | DHCP:10.105.50.22 | Spd:100Mb | Vendor:Intel(R) 82579LM Gigabit Network Connection 
    System Name = J2KV2R1
    System Description = OS:Win 7 Pro | Id:J2KV2R1 | U:AD: USER | Up:000d05h19m16s | Ver: Win NT 6.1.7601 SP 1 
    Capabilities
    Management Address
    Management Address
    IEEE 802.3 - EEE (Energy-Efficient Ethernet)
        1111 111. .... .... = TLV Type: Organization Specific (127)
        .... ...0 0000 0101 = TLV Length: 5
        Organization Unique Code: IEEE 802.3 (0x00120f)
        IEEE 802.3 Subtype: EEE (Energy-Efficient Ethernet) (0x05)
[Malformed Packet: LLDP]
    [Expert Info (Error/Malformed): Malformed Packet (Exception occurred)]
        [Malformed Packet (Exception occurred)]
        [Severity level: Error]
        [Group: Malformed]
```

### In case of bug report: Unexpected behaviour you saw
...

### In case of bug report: Steps to reproduce the problem

With WINPCAP V 4.1.3
Start WinLLDPService
Start Wireshark Version 2.2.4
Capture NIC
LLDP "MALFORMED PACKET" 

### In case of bug report: Additional details
Any errors on Windows Event Log? 
YES Seems that it can not find wpcap.dll. It is in C:\Windows\System32 and in C:\Windows\Syswow64

```
Packet sent failed: System.DllNotFoundException: No se puede cargar el archivo DLL 'wpcap': No se puede encontrar el módulo especificado. (Excepción de HRESULT: 0x8007007E)
   en SharpPcap.WinPcap.SafeNativeMethods.pcap_findalldevs_ex(String source, IntPtr auth, IntPtr& alldevs, StringBuilder errbuf)
   en SharpPcap.WinPcap.WinPcapDeviceList.Devices(String rpcapString, RemoteAuthentication remoteAuthentication)
   en SharpPcap.WinPcap.WinPcapDeviceList.Refresh()
   en SharpPcap.WinPcap.WinPcapDeviceList.get_Instance()
   en SharpPcap.CaptureDeviceList..ctor()
   en SharpPcap.CaptureDeviceList.get_Instance()
   en WinLLDPService.WinLLDP.OpenDevices(List`1 adapters)
   en WinLLDPService.WinLLDP.Run()
   en WinLLDPService.WinLLDPService.SendPacket(Object source, ElapsedEventArgs ea)
```

What .NET framework version? 
Microsoft .NET Framework 4.6.1

Did you try to update msiexec and/or .NET framework? 
YES
Did you try also 32 bit installer on 64 bit system? 
YES
Did you try on different machine(s)?
NO

**Comment by raspi** [comment]
There might be two issues. 

* Invalid characters in NIC name
* DLL not found, but wireshark sees packets but it would be impossible to send packets without working WinPcap

Let's try NIC name issue first: 
Open network and sharing center -> Change adapter settings -> Rename the NIC named "`Conexión de área local`" to contain only ASCII characters like just "`LAN`". Does this resolve the issue?

**Comment by aperezmollo** [comment]
Hi Raspi. Changed NIC name and problem continues. I have taken a look at the packet in WS and found that Management Address Subtype is IPv6 (2). Don't know where the WINLLDP service gets this information from. The LLDP packet from the switch informs Management Address Subtype IPv4 (1).
Any clue?
Thank you

**Comment by aperezmollo** [comment]
Hi Raspi removed IPv6. Issue is the same. 
From the dump of the LLDP packet seems that there is no proper End of LLDPU: End of LLDPU
 0000 000.  . . . .   . . . . = TLV Type: End of LLDPDU (0)
 . . . .   . .  .0 0000 0000 = TLV Length (0)

**Comment by raspi** [comment]
And you've restarted after installing WinPcap?

**Comment by raspi** [comment]
Can you attach the whole lldp frame as .pcap file so I could see it in WS?

**Comment by raspi** [comment]
I just upgraded from Wireshark 2.2.1 to 2.2.4 and same problem started.

**Comment by raspi** [comment]
Ok, I found a fix. I went to official WinPcap page and extracted `WinPcap_4_1_3.exe` with 7-zip and copied  `WinPcap_4_1_3\$SYSDIR\wpcap.dll` to WinLLDPService install directory. Latest WireShark installed ~100KB smaller wpcap.dll to System32. Now the program works. So WireShark is distributing some weird DLL version of WinPcap. Version numbers seems to match.

**Comment by aperezmollo** [comment]
Thank you I will change winpcap and get back to you

**Comment by raspi** [comment]
And of course you need to restart the service after copying the DLL.

**Comment by aperezmollo** [comment]
Hi I have W7 64. Should I reinstall WinLLDP for 64?

**Comment by aperezmollo** [comment]
Hi. I have:
1. Installed WinLLDPService-x64 (1).msi 
2 Downloaded from Winpcap official site 4.1.3
3 Extracted WinPcap_4_1_3.exe with 7-zip and 
4. copied WinPcap_4_1_3\$SYSDIR\wpcap.dll to WinLLDPService install directory: C:\Program Files 
(x86)\WinLLDPService
5 Started WinLLDP Service
6. Run Wireshark. It seems that the packet is malformed. 
I think I have to copy wpcap.dll to other directories like System32 or Syswow64. OK?
Thanks a lot

**Comment by raspi** [comment]
I'm able to replicate the issue and I'm currently trying to find the root cause. I will update this issue when I know more.

**Comment by aperezmollo** [comment]
Thank you!

**Comment by raspi** [comment]
Fixed in new release 17.02.15.2317 https://github.com/raspi/WinLLDPService/releases/tag/v17.02.15.2317

Open new issue if there's new problem(s).
